parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ColumnChunkPageWriteStore.java

```diff
@@ -194,6 +198,10 @@ public long getMemSize() {
 
     public void writeToFileWriter(ParquetFileWriter writer) throws IOException {
       writer.startColumn(path, totalValueCount, compressor.getCodecName());
+      if (bloomFilter != null) {
```

Comment:
So, we write the bloom filters into the row-group. It means we have to seek to the row-group and read from them even if the whole row-group could be dropped based on the bloom filter. Because of the similar issue we decided to the put the column indexes after all row-groups (just before the footer). I don't know if it would be feasible/beneficial for bloom filters as well.

Reply:
According to design, we put bloom filter at beginning of row group.  Not sure how much overhead if we seek at zero and read them. But I can try to put it near to foot to avoid such overhead.

Reply:
Actually, I'm not sure what is the overhead if we read some index data from row group. Does seek read extra data?

Reply:
If you use HDFS as the storage layer and configures parquet properly (aligning the row-groups to HDFS blocks) then you will have different row-groups in different blocks which are usually placed in different hosts. So, seeking in the file might require remote reads which consuming resources on the remote host and the network bandwidth as well. Usually, reading the footer is also a remote read but you have to read the footer anyway. 
See the [column-index design](https://github.com/apache/parquet-format/blob/master/PageIndex.md#technical-approach) for more details.

I would also like to highlight that I am not suggesting we shall go the same direction as we did for the column-indexes. I would only like to encourage you to discuss this topic with the community and include it in the design. It might also require some benchmarking as well.

Reply:
Thanks for the nice explanation. As you point out, it would better to execute some benchmark firstly and let data talk. Will show benchmark data later.